cmd/config/certs.go

```diff
@@ -69,34 +70,63 @@ func ParsePublicCertFile(certFile string) (x509Certs []*x509.Certificate, err er
 	return x509Certs, nil
 }
 
-// GetRootCAs - returns all the root CAs into certPool
-// at the input certsCADir
-func GetRootCAs(certsCAsDir string) (*x509.CertPool, error) {
+// GetRootCAs returns a new CertPool that contains all
+// system root certificates and PEM-encoded certificate(s)
+// found at path.
+//
+// If path is a file, GetRootCAs parses that file and
+// tries to add it to the CertPool. If the file is not
+// a PEM-encoded certificate, GetRootCAs returns an error.
+//
+// If path is a directory, GetRootCAs tries parse each
+// file in that directory as PEM-encoded certificate
+// and adds it to the CertPool. If a file is not a valid
+// PEM-encoded certificate, GetRootCAs ignores that file
+// and continues with the next one.
+func GetRootCAs(path string) (*x509.CertPool, error) {
 	rootCAs, _ := x509.SystemCertPool()
 	if rootCAs == nil {
 		// In some systems (like Windows) system cert pool is
 		// not supported or no certificates are present on the
 		// system - so we create a new cert pool.
 		rootCAs = x509.NewCertPool()
 	}
+	if path == "" {
+		return rootCAs, nil
+	}
 
-	fis, err := ioutil.ReadDir(certsCAsDir)
+	stat, err := os.Stat(path)
 	if err != nil {
-		if os.IsNotExist(err) || os.IsPermission(err) {
-			// Return success if CA's directory is missing or permission denied.
-			err = nil
+		if os.IsNotExist(err) {
+			return rootCAs, nil
```

Comment:
Make sure to ignore the conditions which were ignored previously os.IsPermission()

Reply:
I'm actually not okay with this behavior. If I specify that I want to add a certificate included as RootCA and this does not work because (for some reason) I don't have sufficient permissions, I would expect an error.

Reply:
@aead in certain deployments that's not the case such as k8s has further restrictions on certain directories even if they don't exist. I would expect you to look at original bugs that were fixed and rationale behind it. If you are going to reintroduce them we can't take this PR.